src/components/RightDockedModal.js

```diff
@@ -20,25 +21,73 @@ const propTypes = {
     /* Onyx Props */
     // Url currently in view
     currentURL: PropTypes.string,
+
+    isModalShown: PropTypes.bool.isRequired,
+
+    modalType: PropTypes.string.isRequired,
 };
 
 const defaultProps = {
     route: '',
     currentURL: '',
 };
 
-const RightDockedModal = memo(({
-    route, children, currentURL,
-}) => (
-    <Modal
-        type={CONST.MODAL.MODAL_TYPE.RIGHT_DOCKED}
-        onClose={redirectToLastReport}
-        isVisible={currentURL === route}
-        backgroundColor={themeColors.componentBG}
-    >
-        {children}
-    </Modal>
-));
+class RightDockedModal extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.onClose = this.onClose.bind(this);
+        this.modalHandler = this.modalHandler.bind(this);
+
+        this.state = {
+            isVisible: false,
+        };
+    }
+
+    componentDidMount() {
+        this.modalHandler();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.isModalShown !== this.props.isModalShown || prevProps.currentURL !== this.props.currentURL) {
+            this.modalHandler();
+        }
+    }
+
+    onClose() {
+        const {isModalShown} = this.props;
+        if (isModalShown) {
+            modalHide();
+        }
+        redirectToLastReport();
+    }
+
+    modalHandler() {
+        const {
```

Comment:
Same here, destructing doesn't help make the code any more clear and should be removed.